parlai/tasks/dbll_babi/agents.py

```diff
@@ -55,8 +55,113 @@ def __init__(self, opt, shared=None):
         opt['datafile'] = _path(os.path.join('babi', 'babi1'), params, opt)
         opt['cands_datafile'] = _path(os.path.join('babi', 'babi1'), params,
                                       opt, 'train')
+        self.opt = opt
         super().__init__(opt, shared)
 
+    def setup_data(self, path):
+        if len(self.opt['task'].split(':')) == 4:
+            return self.setup_data_with_feedback(path)
+        else:
+            return super().setup_data(path)
+
+    def setup_data_with_feedback(self, path):
+        """Reads data in the fbdialog format.
```

Comment:
this data doesnt exist (won't be downloaded, though right (right now)?)

Reply:
No, this method only iterates over the dataset. I think the data is downloaded when a new task is created.

Reply:
ok, can we just move this stuff to parlai/projects because it doesn't really fit what we see as a standard ParlAI dataset format. As I said I think it's ok for experiments, but our standard format has to be expecting responses in time sequence format (i.e. you wouldn't get the feedback before you've provided the label!)